packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -252,11 +252,13 @@ class CupertinoSheetTransition extends StatefulWidget {
     final Animation<double> scaleAnimation = curvedAnimation.drive(_kScaleTween);
     curvedAnimation.dispose();
 
-    SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.light);
-
     final bool isDarkMode = CupertinoTheme.brightnessOf(context) == Brightness.dark;
     final Color overlayColor = isDarkMode ? const Color(0xFFc8c8c8) : const Color(0xFF000000);
 
+    SystemChrome.setSystemUIOverlayStyle(
+      isDarkMode ? SystemUiOverlayStyle.light : SystemUiOverlayStyle.dark,
```

Comment:
This isn't quite the correct function, as the sheet needs to set the SystemUIOverlayStyle to light regardless of theme. Since when the sheet is opened, the system UI overlay is always on a black background.

The problem is that this delegated transition is wrapped around the previous route, and does not go away when the sheet is popped. The delegated transition is waiting to be played again. It might be better to move the call to setSystemUIOverlayStyle to one of the sheet's build methods. Or wrap this call here with `if (!secondaryAnimation.isDismissed)` might work.

Reply:
Ohh, yes correct. So, it should be `SystemUIOverlayStyle.light` when sheet is visible. Otherwise inverse of theme.